doc-surrealdb_versioned_docs/version-latest/surrealql/statements/define/event.mdx

```diff
@@ -106,3 +106,37 @@ The `OVERWRITE` clause can be used to define an event and overwrite an existing
 -- Create an EVENT and overwrite if it already exists
 DEFINE EVENT OVERWRITE example ON example THEN {};
 ```
+
+## Events and permissions
+
+Queries inside the event always execute without any permission checks, even when triggered by changes made by a [record user](/docs/surrealdb/security/authentication#record-users). This can be very useful to perform additional checks and changes that involve tables/records that are inaccessible for the record user.
+
+Consider a CREATE query send by the record user that has CREATE access to the `comment` table only:
+
+```surql
+CREATE comment SET
+    post = post:tomatosoup,
+    content = "So delicious!",
+    author = $auth.id
+;
+```
+
+By having the following event defined, SurrealDB will perform the addiitonal checks and changes:
+
+```surql
+DEFINE EVENT on_comment_created ON TABLE comment
+    WHEN $event = "CREATE"
```

Comment:
praise: I like the line splitting here with `WHEN` and `THEN` parts. Maybe we can change the other queries of the page to follow the same pattern?

Reply:
I'll update other examples then. This line splitting is borrowed from the Surreal Deal Store sample :)